app/uk/gov/hmrc/teamsandrepositories/connectors/JenkinsConnector.scala

```diff
@@ -152,12 +152,16 @@ object JenkinsConnector {
 
   import cats.implicits._
 
-  def parse(root: JenkinsRoot, findBuildJobsFunction: String => Future[JenkinsRoot])(implicit ec: ExecutionContext): Future[Seq[JenkinsJob]] = {
-    root.jobs.toList.traverse {
-      case job if isFolder(job)  => findBuildJobsFunction(job.url).flatMap(parse(_, findBuildJobsFunction))
-      case job if isProject(job) => Future(Seq(job))
-      case _                     => Future(Seq.empty)
-    }.map(_.flatten)
+  def parse(root: Option[JenkinsRoot], findBuildJobsFunction: String => Future[Option[JenkinsRoot]])(implicit ec: ExecutionContext): Future[Seq[JenkinsJob]] = {
+    root match {
+      case None => Future(Seq.empty)
+      case Some(value) => value.jobs.toList.traverse {
```

Comment:
should we replace this `traverse` with `foldLeftM` to ensure sequentiality?

Reply:
Sequentiality is not required.

Reply:
If we're making http calls, especially to the same server,I think we should, to not overload jenkins - this will only contribute to longer response times too.